### PR TITLE
Add an option to forgo backups

### PR DESCRIPTION
**Pull request by marcusklaas**
Closes https://github.com/nrc/rustfmt/issues/171.

### CODE REVIEW

**Comment by nrc** [comment]
I had imagined that we would have a WriteMode for replace with no backups, rather than having a config option. That probably requires specifiying the WriteMode on the command line to be useful.

Is there a reason you prefer a config option?

**Comment by marcusklaas** [comment]
Yea, that's most likely a better solution. Hadn't thought of it.

**Comment by marcusklaas** [comment]
Updated!

File: src/filemap.rs [hunk 1]
```diff
@@ -73,40 +73,46 @@ fn write_file(text: &StringBuffer,
...
+                let tmp_file = try!(File::create(&tmp_name));
+                try!(write_system_newlines(tmp_file, text, config));
             }
+
+            try!(fs::rename(filename, bk_name));
+            try!(fs::rename(tmp_name, filename));
         }
+        WriteMode::Overwrite => {
+            // Write text directly over original file.
+            let file = try!(File::create(filename.to_owned()));
```
Comment (nrc) [inline src/filemap.rs]: Should use open instead of create

File: src/bin/rustfmt.rs [hunk 2]
```diff
@@ -13,19 +13,46 @@
...
+{
+    let prefix = "--write-mode=";
+    let mut write_mode = WriteMode::Replace;
+
+    // The NewFile option currently isn't supported because it requires another
+    // parameter, but it can be added later.
+    let args = args.filter(|arg| {
+        if arg.starts_with(prefix) {
+            write_mode = match &arg[prefix.len()..] {
+                "replace" => WriteMode::Replace,
```
Comment (nrc) [inline src/bin/rustfmt.rs]: Could you implement FromString rather than use a match here?

**Comment by nrc** [comment]
r+ modulo the minor comments

File: src/filemap.rs (hunk 1 above)
Comment (marcusklaas) [inline src/filemap.rs]: Sure? `File::open` opens files in read-only mode. `File::create` truncates existing files, which should happen here I think.

**Pull request merged**